src/Lib.Net.Http.WebPush/PushServiceClientException.cs

```diff
@@ -13,15 +14,36 @@ public class PushServiceClientException : Exception
         /// </summary>
         public HttpStatusCode StatusCode { get; }
 
+        /// <summary>
+        /// Gets or sets the HTTP Response Headers of the push service response.
+        /// </summary>
+        public HttpResponseHeaders Headers { get; set; }
+
+        /// <summary>
+        /// Gets or sets the HTTP Response body of the push service response.
+        /// </summary>
+        public string Body { get; set; }
+
+        /// <summary>
+        /// Gets or sets the PushSubscription that initiated the push service response.
+        /// </summary>
+        public PushSubscription PushSubscription { get; set; }
+
         /// <summary>
         /// Creates new instance of <see cref="PushServiceClientException"/> class.
         /// </summary>
-        /// <param name="message">The message that describes the current exception</param>
+        /// <param name="message">The message that describes the current exception.</param>
         /// <param name="statusCode">The status code of the push service response.</param>
-        public PushServiceClientException(string message, HttpStatusCode statusCode)
+        /// <param name="headers">The headers of the push service response.</param>
+        /// <param name="body">The body of the push service response.</param>
+        /// <param name="pushSubscription">The push subscription that initiated the push service response.</param>
+        public PushServiceClientException(string message, HttpStatusCode statusCode, HttpResponseHeaders headers, string body, PushSubscription pushSubscription)
```

Comment:
It would be great if you could add a second constructor instead of modifying the existing one. Unfortunately it's public so changing it is a breaking change.

Reply:
I actually considered creating a new derived "PushDeliveryException" then decided that the existing PushClientServiceException already contained StatusCode so it was implied it was encountered because of communication with the push server.  I think having an exception class with just message as the base would be flexible for future since not all exceptions will be the result of communication with server.

For now, I can refactor to adding a new constructor or just making headers & pushSubscription optional (default null) - let me know which you'd prefer.

Thanks!

Reply:
I prefer new constructor. As it has been proven to me by someone way smarter, adding defaults should be considered binary breaking changes because of how C# resolves them at runtime.

Reply:
np, will do!